src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7647,7 +7590,28 @@ public class FileSystemContentDynamicParametersBase
         /// reading data from the file.
         /// </summary>
         [Parameter]
-        public FileSystemCmdletProviderEncoding Encoding { get; set; } = FileSystemCmdletProviderEncoding.String;
+        [ArgumentToEncodingTransformationAttribute()]
+        [ArgumentCompleter(typeof(EncodingArgumentCompleter))]
```

Comment:
I think we also need a `ValidateNullOrEmpty` here.

Reply:
fixed